packages/components/src/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -0,0 +1,443 @@
+import React from 'react';
+import {SampleTypeModel} from './models';
+import {SampleTypePropertiesPanel} from "./SampleTypePropertiesPanel";
+import {
+    Alert,
+    DomainDesign,
+    DomainDetails,
+    generateId,
+    IDomainField,
+    initQueryGridState,
+    naturalSort,
+    resolveErrorMessage,
+    SAMPLE_TYPE,
+    saveDomain, SCHEMAS,
+    WizardNavButtons
+} from "../../..";
+import DomainForm from "../DomainForm";
+import {IParentOption} from "../../entities/models";
+import {IParentAlias} from "./models";
+import {addDomainField} from "../actions";
+import {initSampleSetSelects,} from "../../samples/actions";
+import {SAMPLE_SET_DISPLAY_TEXT, STICKY_HEADER_HEIGHT} from "../../../constants";
+import {fromJS, List, Map, OrderedMap} from "immutable";
+import {Domain} from "@labkey/api";
+
+const DEFAULT_SAMPLE_FIELD_CONFIG = {
+    required: true,
+    dataType: SAMPLE_TYPE,
+    conceptURI: SAMPLE_TYPE.conceptURI,
+    rangeURI: SAMPLE_TYPE.rangeURI,
+    lookupSchema: 'exp',
+    lookupQuery: 'Materials',
+    lookupType: {...SAMPLE_TYPE},
+    name: 'SampleId',
+} as Partial<IDomainField>;
+
+const NEW_SAMPLE_SET_OPTION: IParentOption = {
+    label: `(Current ${SAMPLE_SET_DISPLAY_TEXT})`,
+    value: "{{this_sample_set}}"
+} as IParentOption;
+
+export const SAMPLE_SET_IMPORT_PREFIX :string = 'materialInputs/';
+export const DATA_CLASS_IMPORT_PREFIX :string = 'dataInputs/';
+const DATA_CLASS_SCHEMA_KEY:string = 'exp/dataclasses';
+const SAMPLEID_RESERVED_ERROR = 'The ' + DEFAULT_SAMPLE_FIELD_CONFIG.name + ' field name is reserved for imported or generated sample ids.';
+
+interface Props {
+    onCancel: () => void
+    onComplete: (response: any) => void
+    beforeFinish?: (formValues: {}) => void
+    initModel: DomainDetails
+    defaultSampleFieldConfig?: Partial<IDomainField>
+    includeDataClasses?: boolean
+
+    //EntityDetailsForm props
+    noun?: string
+    nameExpressionInfoUrl?: string
+    nameExpressionPlaceholder?: string
+
+    //DomainDesigner props
+    containerTop?: number, // This sets the height of the sticky header, default is 60
+    useTheme?: boolean,
+    appPropertiesOnly?: boolean,
+}
+
+interface State {
+    model: SampleTypeModel
+    parentOptions: Array<IParentOption>
+    invalidDomainField: string
+
+    activePanelIndex: number
+
+    submitting: boolean
+    currentPanelIndex: number
+
+    error: React.ReactNode
+}
+
+export class SampleTypeDesigner extends React.PureComponent<Props, State> {
+    private _dirty = false;
+
+    static defaultProps = {
+        nameExpressionPlaceholder: 'Enter a naming pattern (e.g., S-${now:date}-${dailySampleCount})',
+        defaultSampleFieldConfig: DEFAULT_SAMPLE_FIELD_CONFIG,
+        noun: SAMPLE_SET_DISPLAY_TEXT,
+        includeDataClasses: false,
+
+        containerTop: STICKY_HEADER_HEIGHT,
+        useTheme: false,
+        appPropertiesOnly: true,
```

Comment:
I would have expected the default to be the opposite, i.e. `false`, and at that the app would be the one to override that.